contracts/base/UniswapController.sol

```diff
@@ -0,0 +1,61 @@
+pragma solidity 0.5.17;
+
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router01.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../interfaces/ISwapper.sol";
+import "../util/NumbersLib.sol";
+import "./Base.sol";
+
+contract UniswapController is Base, ISwapper {
+    using NumbersLib for *;
+    using SafeERC20 for IERC20;
+
+    IUniswapV2Router01 private constant ROUTER = IUniswapV2Router01(
+        0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D
+    );
+
+    IChainlinkAggregator private chainlink;
+
+    uint256 private minAmountOutPercent;
+    uint256 private maxSecondsToSwap;
+
+    function initialize(uint256 _minAmountOutPercent, uint256 _maxSecondsToSwap)
+        external
+    {
+        require(
+            _minAmountOutPercent >= 8000 && _minAmountOutPercent < 10000,
+            "MIN_AMOUNT_INVALID"
+        );
+        require(
+            _maxSecondsToSwap > 0 && _maxSecondsToSwap <= 3600,
+            "MAX_SECONDS_INVALID"
+        );
```

Comment:
I'm pretty sure we don't need a `deadline` variable. Since we are calling the swap within the same transaction, we can just pass the `now` value and it will still work because the `now` value will never change in the same transaction.